### PR TITLE
Pk1d zchunks

### PR DESCRIPTION
**Pull request by armengau**
New option in pk1d_postprocess: --apply_z_weights (default False).
If set, each FFT chunck at redshift z contributes 2 nearby z bins, with a triangle-shape weight function identical to that used by the QMLE.
Tests with the fujilupe mocks show the impact on the measured P1D is less than 1%.
- The zmin and zmax redshift bins must necessarily be handled differently
- This PR also includes a minor bug fix (header in output P1D FITS file).

### CODE REVIEW

File: py/picca/pk1d/postproc_pk1d.py [hunk 1]
```diff
@@ -349,6 +382,9 @@ def variance_function(snr, amp, zero_point):
                     mean = np.mean(p1d_table[col][select])
```
Comment (iprafols) [inline py/picca/pk1d/postproc_pk1d.py:382]: shouldn't this line be in an else after line 386 in the new version?

File: py/picca/pk1d/postproc_pk1d.py [hunk 2]
```diff
@@ -349,6 +382,9 @@ def variance_function(snr, amp, zero_point):
...
                     # unbiased estimate: num_chunks-1
                     error = np.std(p1d_table[col][select]) / np.sqrt(num_chunks - 1)
```
Comment (iprafols) [inline py/picca/pk1d/postproc_pk1d.py:384]: shouldn't this also be computed differently when apply_z_weights is on?

**Comment by iprafols** [comment]
The changes look fine. I have spotted a couple of lines that potentially will need changes. @armengau  can you review them? Otherwise, I think the PR is ready to merge

File: py/picca/pk1d/postproc_pk1d.py (hunk 1 above)
Comment (armengau) [reply py/picca/pk1d/postproc_pk1d.py]: yes

File: py/picca/pk1d/postproc_pk1d.py (hunk 2 above)
Comment (armengau) [reply py/picca/pk1d/postproc_pk1d.py]: Yes, I computed and implemented the expression now.
Also fixed a bug (missing sqrt) in the other expression when SNR weighting is on

**Comment by armengau** [comment]
I think it's ok now

**Review by iprafols — APPROVED** [review]

**Pull request merged**